.gitattributes

```diff
@@ -1 +1,2 @@
-crates/ra_syntax/tests/data/** -text
+crates/ra_syntax/test_data/** -text eof=LF
+crates/ra_ide_api/src/snapshots/** -text eof=LF
```

Comment:
There's no way this could have worked, [if you read the documentation](https://git-scm.com/docs/gitattributes):
* `eof` doesn't exist, `eol` does
* `eol` only works on `text` files
  * this makes sense, since it's lossy in general, and only text files can accept the lossiness as "CRLF and LF are two ways to represent the same thing" (line endings

If the goal here was to hide files from `git grep`, I'm afraid there is no way to do that without leaving the door open to CRLF (`\r\n`) corruption.

Honestly looking at #1115 I have no idea what's going on. Most likely some buggy tool produced `\r\n` instead of `\n` for line endings, which didn't match the correct (`\n`) ones `git` was dutifully using.

If it's the other way around, then the `autocrlf` setting is being set to the cursed option that generates `\r\n` on the fly from files committed as `\n` (which is not something you should never need or use - only outdated stuff like `notepad.exe` needs it).

Reply:
> only outdated stuff like notepad.exe needs it

Even Notepad supports Unix line endings now: https://www.theregister.com/2018/05/08/windows_notepad_unix_linux_macos/.